207 menu button color

alishaevn opened the pull request:
# story
create ability to customize the collapsed menu button color.

<details>
<summary>detailed explanation</summary>

[move the default theme colors into a module.scss file](https://github.com/scientist-softserv/webstore-component-library/commit/26467543e581dca9280c6ca94a857346c386d400)
this move was done in preparation for allowing the toggle icon rendered
by the "navbar-toggler" class to be styled with the same colors as the
rest of the navbar. the colors exist in a "module.scss" file so that they
can be imported into our ".scss" AND ".jsx" files.

I also updated the colors from greyscale to match those in the webstore
template. greyscale was chosen initially as not to spend too much time
picking a color palette and distracting from component creation. using
color now allows us to see some of the slight color variances better.

[use an editable svg as the navbar toggle icon](https://github.com/scientist-softserv/webstore-component-library/commit/4ebd693be4bee12bb4bac5fad9b00602ca87cfa6)
by default, the background image rendered by the "navbar-toggler-icon"
class is a dark colored hamburger menu icon. since it uses an image, it
was incapable of being dynamically styled with css alone. this commit
replaces the "navbar-toggler-icon" class inside "navbar-toggler" with an
editable svg, allowing us to change the color.

[adjust navbar toggle behavior](https://github.com/scientist-softserv/webstore-component-library/pull/208/commits/a4e7d0405e69dcde73e942fc61e9bafe47709272)
for some reason after adding the custom `ToggleIcon` component to the Navbar,
the screen size that the navbar collapses at changed. It used to collapse
at <992px, but changed to <1232px. changing the `expand` prop to `md`
collapses the navbar at <1008px, which is closer to the original behavior.

- ref: https://react-bootstrap.github.io/docs/components/navbar#responsive-behaviors

[overriding the spread radius for ".navbar-toggler:focus"](https://github.com/scientist-softserv/webstore-component-library/pull/208/commits/69afdc2b59e67581f17b81986d5330414319da54)
I would like to make it the same color as the icon too, but that isn't
possible dynamically with scss
</details>

# demo

https://github.com/scientist-softserv/webstore-component-library/assets/29032869/806cb956-aed7-4932-bd32-db9d86a1720e

alishaevn committed:
move the default theme colors into a module.scss file

this move was done in preparation for allowing the toggle icon rendered
by the "navbar-toggler" class to be styled with the same colors as the
rest of the navbar. the colors exist in a "module.scss" file so that they
can be imported into our ".scss" AND ".jsx" files.

I also updated the colors from greyscale to match those in the webstore
template. greyscale was chosen initially as not to spend too much time
picking a color palette and distracting from component creation. using
color now allows us to see some of the slight color variances better.

- ref: #207

alishaevn committed:
use an editable svg as the navbar toggle icon

by default, the background image rendered by the "navbar-toggler-icon"
class is a dark colored hamburger menu icon. since it uses an image, it
was incapable of being dynamically styled with css alone. this commit
replaces the "navbar-toggler-icon" class inside "navbar-toggler" with an
editable svg, allowing us to change the color.

- ref: #207

alishaevn committed:
adjust navbar toggle behavior

for some reason after adding the custom ToggleIcon component to the Navbar,
the screen size that the navbar collapses at changed. It used to collapse
at <992px, but changed to <1232px. changing the `expand` prop to `md`
collapses the navbar at <1008px, which is closer to the original behavior.

ref: https://react-bootstrap.github.io/docs/components/navbar#responsive-behaviors

alishaevn committed:
slight refactor of ToggleIcon

alishaevn committed:
Merge branch 'main' into 207-menu-button-color

alishaevn committed:
overriding the spread radius for ".navbar-toggler:focus"

I would like to make it the same color as the icon too, but that isn't
possible dynamically with scss

alishaevn commented on src/compounds/Header/styles.scss:
```diff
@@ -1,3 +1,11 @@
   height: 45px;
 }
+
+.navbar-toggler:focus {
+  text-decoration: none;
+  outline: 0;
+  // overriding the below to lessen the spread radius
+  // I would like to make it the same color as the icon, but that isn't possible dynamically with scss
+  box-shadow: 0 0 0 1px;
+}
```
### before
<img width="732" alt="image" src="https://github.com/scientist-softserv/webstore-component-library/assets/29032869/28697b58-0328-4e15-b5d3-fe049098ecca">

### after
<img width="730" alt="image" src="https://github.com/scientist-softserv/webstore-component-library/assets/29032869/30bee111-093b-4978-b1d8-70f6baa8a6a2">

labradford reviewed (commented):
If I try to set the color to a rgb or hex value, there isn't a change. It only changes if I type in one of the color names in the default colors list. If that is intentional, could you make the selection a drop down list of colors rather than a color picker?

alishaevn commented:
> If I try to set the color to a rgb or hex value, there isn't a change. It only changes if I type in one of the color names in the default colors list. If that is intentional, could you make the selection a drop down list of colors rather than a color picker?

yes, passing a string value that equates to one of the default colors is the intended behavior. it's because it comes in that way [from the webstore](https://github.com/scientist-softserv/webstore/blob/main/utils/theme/variables.js#L18) with the way we're using the bootstrap theme there.
I'm unsure off the top of my head if I can make the storybook color picker a dropdown....but I can look into it.

alishaevn committed:
more accurate propType definition of Header.linkColor

alishaevn committed:
revert the Navbar.Brand to only 50% width

for some reason, with the `w-75` class, the "Sign In" text in the Header
was being shown on two lines. Not sure why, when this wasn't happening
yesterday. retested with tall and wide logos to make sure they still
are contained in the given area and don't distort.
